.travis.yml

```diff
@@ -21,60 +21,67 @@
-      env: BUILD_TARGET="linux-auto-gcc-lint" CC="gcc"
-      os: linux
-      compiler: gcc
-    - name: "Linux with Defaults against clang/LLVM"
-      env: BUILD_TARGET="linux-auto-clang" CC="clang"
-      os: linux
-      compiler: clang
-    - name: "Linux with LWIP against clang/LLVM"
-      env: BUILD_TARGET="linux-lwip-clang" CC="clang"
-      os: linux
-      compiler: clang
-    - name: "OS X with Defaults against clang/LLVM"
-      env: BUILD_TARGET="osx-auto-clang" CC="clang"
-      os: osx
-      compiler: clang
-    - name: "OS X with LwIP against clang/LLVM"
-      env: BUILD_TARGET="osx-lwip-clang" CC="clang"
-      os: osx
-      compiler: clang
-    - name: "ESP32"
-      env: BUILD_TARGET="esp32"
-    - name: "Linux with Defaults against GCC Functional and Unit Tests"
-      env: BUILD_TARGET="linux-auto-gcc-check" CC="gcc"
-      os: linux
-      compiler: gcc
-    - name: "Linux with LwIP against GCC Functional and Unit Tests"
-      env: BUILD_TARGET="linux-lwip-gcc-check" CC="gcc"
-      os: linux
-      compiler: gcc
-    - name: "Linux with Defaults against GCC Network and Service Tests"
-      os: linux
-      compiler: gcc
-      env: BUILD_TARGET="linux-auto-gcc-check-happy"
-    - name: "Linux with LwIP against GCC Network and Service Tests"
-      os: linux
-      compiler: gcc
-      env: BUILD_TARGET="linux-lwip-gcc-check-happy"
-
+  - name: Linux with Defaults against GCC Distribution Sanity Check
+    env: BUILD_TARGET="linux-auto-gcc-distcheck" CC="gcc"
+    os: linux
+    compiler: gcc
+  - name: Linux with Defaults against GCC Coding Style Check
+    env: BUILD_TARGET="linux-auto-gcc-lint" CC="gcc"
+    os: linux
+    compiler: gcc
+  - name: Linux with Defaults against clang/LLVM
+    env: BUILD_TARGET="linux-auto-clang" CC="clang"
+    os: linux
+    compiler: clang
+  - name: Linux with LWIP against clang/LLVM
+    env: BUILD_TARGET="linux-lwip-clang" CC="clang"
+    os: linux
+    compiler: clang
+  - name: OS X with Defaults against clang/LLVM
+    env: BUILD_TARGET="osx-auto-clang" CC="clang"
+    os: osx
+    compiler: clang
+  - name: OS X with LwIP against clang/LLVM
+    env: BUILD_TARGET="osx-lwip-clang" CC="clang"
+    os: osx
+    compiler: clang
+  - name: ESP32
+    env: BUILD_TARGET="esp32"
+  - name: Linux with Defaults against GCC Functional and Unit Tests
+    env: BUILD_TARGET="linux-auto-gcc-check" CC="gcc"
+    os: linux
+    compiler: gcc
+  - name: Linux with LwIP against GCC Functional and Unit Tests
+    env: BUILD_TARGET="linux-lwip-gcc-check" CC="gcc"
+    os: linux
+    compiler: gcc
+  - name: Linux with Defaults against GCC Network and Service Tests
+    os: linux
+    compiler: gcc
+    env: BUILD_TARGET="linux-auto-gcc-check-happy" TRAVIS_TAG="true"
+  - name: Linux with LwIP against GCC Network and Service Tests
+    os: linux
+    compiler: gcc
+    env: BUILD_TARGET="linux-lwip-gcc-check-happy" TRAVIS_TAG="true"
+deploy:
```

Comment:
New line before this line. In a few short weeks, all of this is going to be tribal knowledge. In fact, I couldn't describe a single line of what's going on below. This whole block needs a substantial preamble about what's going on here and why.

Reply:
Fixed this one, please check again.

Reply:
Thanks for adding the documentation and removing white space changes.